frontend/packages/console-dynamic-plugin-sdk/src/extensions/topology.ts

```diff
@@ -39,7 +39,7 @@ export type TopologyDataModelFactory = ExtensionDeclaration<
     /** Priority for the factory */
     priority: number;
     /** Resources to be fetched from useK8sWatchResources hook. */
-    resources?: WatchK8sResourcesGeneric;
+    resources?: CodeRef<WatchResourcesHook>;
```

Comment:
Better to use `ExtensionHook` here. See my other comment.

Reply:
Updated to use `ExtensionHook`.